packages/core/src/utils.ts

```diff
@@ -237,6 +237,8 @@ export function showErrorForDeprecatedField(
   );
 }
 
+export type OptionalIntersectionTypes<T, T2> = Partial<Omit<T, keyof T2>> & T2;
```

Comment:
Could you leave a comment or simplify the type?

Reply:
It's a little bit complicated and hard to read, I'll remove this generic.